ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/validation/CustomValidators.java

```diff
@@ -0,0 +1,219 @@
+    public void validate(
+        final PreparedStatement statement,
+        final KsqlExecutionContext context,
+        final KsqlConfig ksqlConfig,
+        final Map<String, Object> propertyOverrides) {
+      throw new KsqlRestException(Errors.queryEndpoint(statement.getStatementText()));
+    }
+  },
+
+  SET_PROPERTY(SetProperty.class) {
+    @SuppressFBWarnings("RV_RETURN_VALUE_IGNORED_INFERRED")
+    @Override
+    public void validate(
+        final PreparedStatement statement,
+        final KsqlExecutionContext context,
+        final KsqlConfig ksqlConfig,
+        final Map<String, Object> propertyOverrides
+    ) {
+      final SetProperty setProperty = (SetProperty) statement.getStatement();
+      CustomValidators.throwIfUnknownProperty(
+          setProperty.getPropertyName(),
+          statement.getStatementText()
+      );
+
+      try {
+        ksqlConfig.cloneWithPropertyOverwrite(ImmutableMap.of(
+            setProperty.getPropertyName(),
+            setProperty.getPropertyValue()
+        ));
+      } catch (final Exception e) {
+        throw new KsqlStatementException(
+            e.getMessage(), statement.getStatementText(), e.getCause());
+      }
+
+      context.execute(statement, ksqlConfig, propertyOverrides);
+    }
+  },
+
+  UNSET_PROPERTY(UnsetProperty.class) {
+    @Override
+    public void validate(
+        final PreparedStatement statement,
+        final KsqlExecutionContext context,
+        final KsqlConfig ksqlConfig,
+        final Map<String, Object> propertyOverrides
+    ) {
+      final UnsetProperty unsetProperty = (UnsetProperty) statement.getStatement();
+      CustomValidators.throwIfUnknownProperty(
+          unsetProperty.getPropertyName(),
+          statement.getStatementText()
+      );
+      context.execute(statement, ksqlConfig, propertyOverrides);
+    }
+  },
+
+  SHOW_COLUMNS(ShowColumns.class) {
+    @Override
+    public void validate(
+        final PreparedStatement statement,
+        final KsqlExecutionContext context,
+        final KsqlConfig ksqlConfig,
+        final Map<String, Object> propertyOverrides
+    ) {
+      final ShowColumns showColumns = (ShowColumns) statement.getStatement();
+      final String name = showColumns.getTable().getSuffix();
+
+      if (showColumns.isTopic()) {
+        final KsqlTopic ksqlTopic = context.getMetaStore().getTopic(name);
+        if (ksqlTopic == null) {
+          throw new KsqlStatementException(
+              "Could not find Topic in the Metastore: " + name,
+              statement.getStatementText());
+        }
+      } else {
+        final StructuredDataSource dataSource = context.getMetaStore().getSource(name);
+        if (dataSource == null) {
+          throw new KsqlStatementException(
+              "Could not find STREAM/TABLE in the Metastore: " + name,
+              statement.getStatementText());
+        }
+      }
+    }
+  },
+
+  EXPLAIN(Explain.class) {
+    @SuppressWarnings("unchecked")
```

Comment:
Validation should just call the same logic as the executor.

I know you've pulled the common logic out into `ExplainExecutor`, but I'd just make `ExplainExecutor` a statement executor / validator. (They should share the same interface).